### PR TITLE
2093. Cells stuck running

### PR DESCRIPTION
**Pull request by maxyeg**
Fix for issue #2093

### CODE REVIEW

**Comment by ildipo** [comment]
Hi Maxy, I think this should be done when you save the notebook: do not save cells 'under evaluation'

**Comment by scottdraves** [comment]
sorry maxy, davide is right, but then the error message saved should be a different from a normal "interrupted" state, like "interrupted, unavailable at save time".

File: core/src/main/web/app/mainapp/mainapp.js [hunk 1]
```diff
@@ -208,6 +208,19 @@
             showLoadingStatusMessage("Loading notebook");
             $scope.loading = true;
 
+            //Clear previous evaluation in progress state.
+            //Set state to Error with message "Interrupted."
+            if (notebookModel && notebookModel.cells) {
```
Comment (samccone) [inline core/src/main/web/app/mainapp/mainapp.js]: Thoughts on moving this into the`bk.notebookCellModelManager` helper?

File: core/src/main/web/app/mainapp/services/notebookcellmodelmanager.js [hunk 2]
```diff
@@ -453,6 +453,18 @@
...
         }
       },
+      clearEvaluatingCells: function () {
+        if (cells) {
+          for (var i = 0; i < cells.length; i++) {
+            var currentCell = cells[i];
+            if (currentCell && currentCell.output && currentCell.output.result
+                && currentCell.output.result.innertype === 'Progress') {
+              currentCell.output.result.innertype = 'Error';
+              currentCell.output.result.object = 'Interrupted.'
```
Comment (scottdraves) [inline core/src/main/web/app/mainapp/services/notebookcellmodelmanager.js]: error message should read:

```
Interrupted, saved while running
```

**Comment by scottdraves** [comment]
This isn't quite right because if a cell is running, and you save the notebook, it actually stops the cell,
and puts the "Interrupted" error state in the notebook.

It should only change the model of the notebook that is being saved to disk, it should not alter the current model in RAM.

**Comment by maxyeg** [comment]
Please, review only the last commit: https://github.com/twosigma/beaker-notebook/commit/d4b751247384b3f38dce4ad6c9fc109ce2af4fab
Rebasing against master brought many new commits to this branch :(
Please, tell me if it'll be better to create new PR.

**Comment by scottdraves** [comment]
yes new PR please

**Pull request closed without merging**